cmake/get_info.cmake

```diff
@@ -0,0 +1,302 @@
+SET(detailed_log "${CMAKE_BINARY_DIR}/detailed.log")
+SET(minimal_log  "${CMAKE_BINARY_DIR}/minimal.log")
+FILE(REMOVE ${detailed_log} ${minimal_log})
+
+MACRO(_both)
+  # Write to both log files:
+  FILE(APPEND ${detailed_log} "${ARGN}")
+  FILE(APPEND ${minimal_log} "${ARGN}")
+ENDMACRO()
+
+MACRO(_detailed)
+  # Only write to detailed.log:
+  FILE(APPEND ${detailed_log} "${ARGN}")
+ENDMACRO()
+
+MACRO(_minimal)
+  # Only write to minimal.log:
+  FILE(APPEND ${minimal_log} "${ARGN}")
+ENDMACRO()
+
+FUNCTION(build_type_spec log_type var_name)
+    if(CMAKE_BUILD_TYPE MATCHES "Release")
```

Comment:
I am not sure `var_name_UPPER{CMAKE_BUILD_TYPE}` is general rule. 
you maybe use string(TOUPPER "$CMAKE_BUILD_TYPE" <output variable>) to generalize this function

Reply:
For alignment, you maybe need to use `string(SUBSTRING...` to get the same length result.
put many spaces after `${var_name}_${TYPE}` (maybe 40 spaces), use `SUBSTRING` to get fixed-length result and then append `${${var_string}}`
Current version does not align it well.
```
#        CMAKE_CUDA_COMPILER_VERSION:            9.2.148
#        CMAKE_CUDA_FLAGS_RELEASE                   -O3 -DNDEBUG 
```

```
set(upd_string "")